backend/src/org/commcare/session/SessionFrame.java

```diff
@@ -95,12 +98,10 @@ public String getFrameId() {
      * This snapshot can be referenced later to compare the eventual state
      * of the frame to an earlier point
      */
-    public void captureSnapshot() {
-        synchronized (steps) {
```

Comment:
I didn't really remove it, I moved it to the method signature.
The reason I did that is because all the usages to `synchronized (steps)` wrapped the entire method, which means it is equivalent to making the whole method `synchronized`